patches/server/0924-Add-drops-to-shear-events.patch

```diff
@@ -233,33 +233,75 @@ index 2de1a2f666da9db1832907e1651dbff948e37252..5c2ed3c39c8eb850f3be1e2ea5b5a7ea
  
      }
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Bogged.java b/src/main/java/net/minecraft/world/entity/monster/Bogged.java
-index 8420d75865b86e1d8afea2527b5521cac184e8b1..6e290d67b00c88ecd2cf2ce5f612f52ebda9e280 100644
+index dc6230458e09f7555eee7f6a567ff60ad454666b..601a1460e92754b05b337d3f53f84ac2d85f09c8 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Bogged.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Bogged.java
-@@ -135,6 +135,15 @@ public class Bogged extends AbstractSkeleton implements Shearable {
+@@ -80,12 +80,19 @@ public class Bogged extends AbstractSkeleton implements Shearable {
+ 
+         if (itemstack.is(Items.SHEARS) && this.readyForShearing()) {
+             // CraftBukkit start
+-            if (!org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerShearEntityEvent(player, this, itemstack, hand)) {
+-                this.getEntityData().markDirty(Bogged.DATA_SHEARED); // CraftBukkit - mark dirty to restore sheared state to clients
+-                return InteractionResult.PASS;
++            // Paper start - expose drops in event
++            java.util.List<net.minecraft.world.item.ItemStack> drops = generateDefaultDrops();
++            final org.bukkit.event.player.PlayerShearEntityEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerShearEntityEvent(player, this, itemstack, hand, drops);
++            if (event != null) {
++                if (event.isCancelled()) {
++                    this.getEntityData().markDirty(Bogged.DATA_SHEARED); // CraftBukkit - mark dirty to restore sheared state to clients
```

Comment:
using Entity#resendPossiblyDesyncedEntityData would be better

Reply:
It is yea 👍